### PR TITLE
Added unit test for existing business post and comment models

### PR DESCRIPTION
**Pull request by DamionSilver**
Originally, unit testing was mistaken for integration testing. We switched the directories, and we realized we had no unit testing for the business logic. We implemented Unit testing for the comment and post models inside the unit testing directory.

### CODE REVIEW

File: Fakebook.Posts/Fakebook.Posts.Domain/Models/Comment.cs [hunk 1]
```diff
@@ -7,16 +7,48 @@ namespace Fakebook.Posts.Domain.Models
     public class Comment
     {
         public int Id { get; set; }
-        public string Content { get; set; }
+        private string _Content;
```
Comment (escalonn) [inline Fakebook.Posts/Fakebook.Posts.Domain/Models/Comment.cs:10]: keep consistent with project naming convention - private fields should start with underscore and lowercase letter

File: Fakebook.Posts/Fakebook.Posts.Domain/Models/Comment.cs [hunk 2]
```diff
@@ -7,16 +7,48 @@ namespace Fakebook.Posts.Domain.Models
...
+        public string UserEmail{
+            get{
+                return _UserEmail;
+            }
+            set{
+                if(value.Contains('@')){
+                    _UserEmail = value;
+                }
+                else{
+                    throw new ArgumentException("A correct email format is required.");
```
Comment (escalonn) [inline Fakebook.Posts/Fakebook.Posts.Domain/Models/Comment.cs:26]: use the parameter name argument with argumentexception as well - nameof(value) in the case of a property setter like this.
check against null first as well with argumentnullexception.
i guess it's fine to move the validation out of the constructor into the property setters, but do the setters need to be public? i guess for the content, sure, we could edit that. but the user email?

File: Fakebook.Posts/Fakebook.Posts.UnitTests/PostTests.cs [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+         [Fact]
+        public void Post_EmailIsNotNull(){
+            //arrange
+                const string Content = "This is some content";
+                const string userEmail = "damion.silver@test.com";
+            //act
+            Post newcomment = new Post(userEmail, Content);
+
+            //assert
+                Assert.NotNull(newcomment.UserEmail);
```
Comment (escalonn) [inline Fakebook.Posts/Fakebook.Posts.UnitTests/PostTests.cs:35]: weird indentation throughout this PR. stick to the preexisting style. it's easy, every editor has a command for it

**Review by escalonn — CHANGES_REQUESTED** [review]
implementing behavior and tests is supposed to be done in separate tdd-fail and tdd-pass pull requests to a feature branch, not all together in one PR to main branch.

File: Fakebook.Posts/Fakebook.Posts.Domain/Models/Comment.cs [hunk 4]
```diff
@@ -13,10 +13,16 @@ public class Comment
...
+
+
+
+
         public Comment(string userEmail, string content)
         {
             if (string.IsNullOrWhiteSpace(userEmail)) throw new ArgumentException("User email is required.", nameof(userEmail));
             if (string.IsNullOrWhiteSpace(content)) throw new ArgumentException("Post content is required.", nameof(content));
+            if (!userEmail.Contains("@")) throw new ArgumentException("Post content is required.", nameof(content));
+
```
Comment (escalonn) [inline Fakebook.Posts/Fakebook.Posts.Domain/Models/Comment.cs:25]: this exception doesnt make sense.
and the whitespace being added above should not be there

File: Fakebook.Posts/Fakebook.Posts.Domain/Models/Post.cs [hunk 5]
```diff
@@ -17,6 +17,8 @@ public Post(string userEmail, string content)
         {
             if (string.IsNullOrWhiteSpace(userEmail)) throw new ArgumentException("User email is required.", nameof(userEmail));
             if (string.IsNullOrWhiteSpace(content)) throw new ArgumentException("Post content is required.", nameof(content));
+            if (!userEmail.Contains("@")) throw new ArgumentException("Post content is required.", nameof(content));
+
```
Comment (escalonn) [inline Fakebook.Posts/Fakebook.Posts.Domain/Models/Post.cs:21]: this exception doesnt make sense

File: Fakebook.Posts/Fakebook.Posts.IntegrationTests/Controllers/DeleteCommentTests.cs [hunk 6]
```diff
@@ -11,7 +11,7 @@
 using Moq;
 using Xunit;
 
-namespace Fakebook.Posts.UnitTests.Controllers
+namespace Fakebook.Posts.IntegrationTests.Controllers
```
Comment (escalonn) [inline Fakebook.Posts/Fakebook.Posts.IntegrationTests/Controllers/DeleteCommentTests.cs:14]: this branch has unrelated changes in it from a different pull request. that's a result of either where this branch was started from, or maybe where you merged into it from. two general solutions to that kind of problem - extricate those changes from this branch, which can have some pitfalls to it, or, this branch just has to wait until the other branch is merged, at which point these changes will disappear from the perspective of the PR.

**Review by escalonn — CHANGES_REQUESTED** [review]
this is not following our tdd process. check with team lead.

**Comment by nosyrbllewe** [comment]
Created new branch and pull request: https://github.com/2011-fakebook-project3/posts/pull/138

**Pull request closed without merging**